sentry-types/src/dsn.rs

```diff
@@ -82,17 +82,33 @@ impl Dsn {
         auth_from_dsn_and_client(self, client_agent)
     }
 
-    /// Returns the submission API URL.
-    pub fn store_api_url(&self) -> Url {
+    fn api_url(&self, endpoint: &str) -> Url {
         use std::fmt::Write;
         let mut buf = format!("{}://{}", self.scheme(), self.host());
         if self.port() != self.scheme.default_port() {
             write!(&mut buf, ":{}", self.port()).unwrap();
         }
-        write!(&mut buf, "{}api/{}/store/", self.path, self.project_id()).unwrap();
+        write!(
+            &mut buf,
+            "{}api/{}/{}/",
+            self.path,
+            self.project_id(),
+            endpoint
+        )
+        .unwrap();
         Url::parse(&buf).unwrap()
     }
 
+    /// Returns the submission API URL.
+    pub fn store_api_url(&self) -> Url {
+        self.api_url("store")
```

Comment:
This seems a bit weird.  I wonder if it might be better off storing the base `Url` struct in the DSN and then use its `join()` method for these.

Reply:
lol, the `Url::join` implementation invokes the whole parse machinery anyway so it really doesn't matter.  this is probably cleaner from the dsn's point of view then